File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-riskdef-5-4/SensitivityDefinition.cpp
// SensitivityDefinition.cpp 
#include "SensitivityDefinition.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

SensitivityDefinition::SensitivityDefinition(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //nameNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* nameNode = xmlNode->FirstChildElement("name");

   if(nameNode){nameIsNull_ = false;}
   else{nameIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- nameNode , address : " << nameNode << std::endl;
   #endif
   if(nameNode)
   {
      if(nameNode->Attribute("href") || nameNode->Attribute("id"))
      {
          if(nameNode->Attribute("id"))
          {
             nameIDRef_ = nameNode->Attribute("id");
             name_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(nameNode));
             name_->setID(nameIDRef_);
             IDManager::instance().setID(nameIDRef_,name_);
          }
          else if(nameNode->Attribute("href")) { nameIDRef_ = nameNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { name_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(nameNode));}
   }

   //valuationScenarioReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* valuationScenarioReferenceNode = xmlNode->FirstChildElement("valuationScenarioReference");

   if(valuationScenarioReferenceNode){valuationScenarioReferenceIsNull_ = false;}
   else{valuationScenarioReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- valuationScenarioReferenceNode , address : " << valuationScenarioReferenceNode << std::endl;
   #endif
   if(valuationScenarioReferenceNode)
   {
      if(valuationScenarioReferenceNode->Attribute("href") || valuationScenarioReferenceNode->Attribute("id"))
      {
          if(valuationScenarioReferenceNode->Attribute("id"))
          {
             valuationScenarioReferenceIDRef_ = valuationScenarioReferenceNode->Attribute("id");
             valuationScenarioReference_ = boost::shared_ptr<ValuationScenarioReference>(new ValuationScenarioReference(valuationScenarioReferenceNode));
             valuationScenarioReference_->setID(valuationScenarioReferenceIDRef_);
             IDManager::instance().setID(valuationScenarioReferenceIDRef_,valuationScenarioReference_);
          }
          else if(valuationScenarioReferenceNode->Attribute("href")) { valuationScenarioReferenceIDRef_ = valuationScenarioReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { valuationScenarioReference_ = boost::shared_ptr<ValuationScenarioReference>(new ValuationScenarioReference(valuationScenarioReferenceNode));}
   }

   //partialDerivativeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* partialDerivativeNode = xmlNode->FirstChildElement("partialDerivative");

   if(partialDerivativeNode){partialDerivativeIsNull_ = false;}
   else{partialDerivativeIsNull_ = true;}

   if(partialDerivativeNode)
   {
      for(partialDerivativeNode; partialDerivativeNode; partialDerivativeNode = partialDerivativeNode->NextSiblingElement("partialDerivative")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partialDerivativeNode , address : " << partialDerivativeNode << std::endl;
          #endif
          if(partialDerivativeNode->Attribute("href") || partialDerivativeNode->Attribute("id"))
          {
              if(partialDerivativeNode->Attribute("id"))
              {
                  partialDerivativeIDRef_ = partialDerivativeNode->Attribute("id");
                  partialDerivative_.push_back(boost::shared_ptr<PricingParameterDerivative>(new PricingParameterDerivative(partialDerivativeNode)));
                  partialDerivative_.back()->setID(partialDerivativeIDRef_);
                  IDManager::instance().setID(partialDerivativeIDRef_, partialDerivative_.back());
              }
              else if(partialDerivativeNode->Attribute("href")) { partialDerivativeIDRef_ = partialDerivativeNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { partialDerivative_.push_back(boost::shared_ptr<PricingParameterDerivative>(new PricingParameterDerivative(partialDerivativeNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partialDerivativeNode , address : " << partialDerivativeNode << std::endl;
       #endif
   }

   //formulaNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* formulaNode = xmlNode->FirstChildElement("formula");

   if(formulaNode){formulaIsNull_ = false;}
   else{formulaIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- formulaNode , address : " << formulaNode << std::endl;
   #endif
   if(formulaNode)
   {
      if(formulaNode->Attribute("href") || formulaNode->Attribute("id"))
      {
          if(formulaNode->Attribute("id"))
          {
             formulaIDRef_ = formulaNode->Attribute("id");
             formula_ = boost::shared_ptr<DerivativeFormula>(new DerivativeFormula(formulaNode));
             formula_->setID(formulaIDRef_);
             IDManager::instance().setID(formulaIDRef_,formula_);
          }
          else if(formulaNode->Attribute("href")) { formulaIDRef_ = formulaNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { formula_ = boost::shared_ptr<DerivativeFormula>(new DerivativeFormula(formulaNode));}
   }

   //termNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* termNode = xmlNode->FirstChildElement("term");

   if(termNode){termIsNull_ = false;}
   else{termIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- termNode , address : " << termNode << std::endl;
   #endif
   if(termNode)
   {
      if(termNode->Attribute("href") || termNode->Attribute("id"))
      {
          if(termNode->Attribute("id"))
          {
             termIDRef_ = termNode->Attribute("id");
             term_ = boost::shared_ptr<TimeDimension>(new TimeDimension(termNode));
             term_->setID(termIDRef_);
             IDManager::instance().setID(termIDRef_,term_);
          }
          else if(termNode->Attribute("href")) { termIDRef_ = termNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { term_ = boost::shared_ptr<TimeDimension>(new TimeDimension(termNode));}
   }

   //coordinateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* coordinateNode = xmlNode->FirstChildElement("coordinate");

   if(coordinateNode){coordinateIsNull_ = false;}
   else{coordinateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- coordinateNode , address : " << coordinateNode << std::endl;
   #endif
   if(coordinateNode)
   {
      if(coordinateNode->Attribute("href") || coordinateNode->Attribute("id"))
      {
          if(coordinateNode->Attribute("id"))
          {
             coordinateIDRef_ = coordinateNode->Attribute("id");
             coordinate_ = boost::shared_ptr<PricingDataPointCoordinate>(new PricingDataPointCoordinate(coordinateNode));
             coordinate_->setID(coordinateIDRef_);
             IDManager::instance().setID(coordinateIDRef_,coordinate_);
          }
          else if(coordinateNode->Attribute("href")) { coordinateIDRef_ = coordinateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { coordinate_ = boost::shared_ptr<PricingDataPointCoordinate>(new PricingDataPointCoordinate(coordinateNode));}
   }

   //coordinateReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* coordinateReferenceNode = xmlNode->FirstChildElement("coordinateReference");

   if(coordinateReferenceNode){coordinateReferenceIsNull_ = false;}
   else{coordinateReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- coordinateReferenceNode , address : " << coordinateReferenceNode << std::endl;
   #endif
   if(coordinateReferenceNode)
   {
      if(coordinateReferenceNode->Attribute("href") || coordinateReferenceNode->Attribute("id"))
      {
          if(coordinateReferenceNode->Attribute("id"))
          {
             coordinateReferenceIDRef_ = coordinateReferenceNode->Attribute("id");
             coordinateReference_ = boost::shared_ptr<PricingDataPointCoordinateReference>(new PricingDataPointCoordinateReference(coordinateReferenceNode));
             coordinateReference_->setID(coordinateReferenceIDRef_);
             IDManager::instance().setID(coordinateReferenceIDRef_,coordinateReference_);
          }
          else if(coordinateReferenceNode->Attribute("href")) { coordinateReferenceIDRef_ = coordinateReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { coordinateReference_ = boost::shared_ptr<PricingDataPointCoordinateReference>(new PricingDataPointCoordinateReference(coordinateReferenceNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeString> SensitivityDefinition::getName()
{
   if(!this->nameIsNull_){
        if(nameIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeString>(IDManager::instance().getID(nameIDRef_));
        }else{
             return this->name_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeString>();
   }
}
boost::shared_ptr<ValuationScenarioReference> SensitivityDefinition::getValuationScenarioReference()
{
   if(!this->valuationScenarioReferenceIsNull_){
        if(valuationScenarioReferenceIDRef_ != ""){
             return boost::shared_static_cast<ValuationScenarioReference>(IDManager::instance().getID(valuationScenarioReferenceIDRef_));
        }else{
             return this->valuationScenarioReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ValuationScenarioReference>();
   }
}
std::vector<boost::shared_ptr<PricingParameterDerivative>> SensitivityDefinition::getPartialDerivative()
{
   if(!this->partialDerivativeIsNull_){
        if(partialDerivativeIDRef_ != ""){
             return this->partialDerivative_;
        }else{
             return this->partialDerivative_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<PricingParameterDerivative>>();
   }
}
boost::shared_ptr<DerivativeFormula> SensitivityDefinition::getFormula()
{
   if(!this->formulaIsNull_){
        if(formulaIDRef_ != ""){
             return boost::shared_static_cast<DerivativeFormula>(IDManager::instance().getID(formulaIDRef_));
        }else{
             return this->formula_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<DerivativeFormula>();
   }
}
boost::shared_ptr<TimeDimension> SensitivityDefinition::getTerm()
{
   if(!this->termIsNull_){
        if(termIDRef_ != ""){
             return boost::shared_static_cast<TimeDimension>(IDManager::instance().getID(termIDRef_));
        }else{
             return this->term_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<TimeDimension>();
   }
}
boost::shared_ptr<PricingDataPointCoordinate> SensitivityDefinition::getCoordinate()
{
   if(!this->coordinateIsNull_){
        if(coordinateIDRef_ != ""){
             return boost::shared_static_cast<PricingDataPointCoordinate>(IDManager::instance().getID(coordinateIDRef_));
        }else{
             return this->coordinate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PricingDataPointCoordinate>();
   }
}
boost::shared_ptr<PricingDataPointCoordinateReference> SensitivityDefinition::getCoordinateReference()
{
   if(!this->coordinateReferenceIsNull_){
        if(coordinateReferenceIDRef_ != ""){
             return boost::shared_static_cast<PricingDataPointCoordinateReference>(IDManager::instance().getID(coordinateReferenceIDRef_));
        }else{
             return this->coordinateReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PricingDataPointCoordinateReference>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-riskdef-5-4/SensitivityDefinition.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class SensitivityDefinition : ISerialized
    {
        public SensitivityDefinition(XmlNode xmlNode)
        {
            XmlNode nameNode = xmlNode.SelectSingleNode("name");
            
            if (nameNode != null)
            {
                if (nameNode.Attributes["href"] != null || nameNode.Attributes["id"] != null) 
                {
                    if (nameNode.Attributes["id"] != null) 
                    {
                        nameIDRef_ = nameNode.Attributes["id"].Value;
                        XsdTypeString ob = new XsdTypeString(nameNode);
                        IDManager.SetID(nameIDRef_, ob);
                    }
                    else if (nameNode.Attributes["href"] != null)
                    {
                        nameIDRef_ = nameNode.Attributes["href"].Value;
                    }
                    else
                    {
                        name_ = new XsdTypeString(nameNode);
                    }
                }
                else
                {
                    name_ = new XsdTypeString(nameNode);
                }
            }
            
        
            XmlNode valuationScenarioReferenceNode = xmlNode.SelectSingleNode("valuationScenarioReference");
            
            if (valuationScenarioReferenceNode != null)
            {
                if (valuationScenarioReferenceNode.Attributes["href"] != null || valuationScenarioReferenceNode.Attributes["id"] != null) 
                {
                    if (valuationScenarioReferenceNode.Attributes["id"] != null) 
                    {
                        valuationScenarioReferenceIDRef_ = valuationScenarioReferenceNode.Attributes["id"].Value;
                        ValuationScenarioReference ob = new ValuationScenarioReference(valuationScenarioReferenceNode);
                        IDManager.SetID(valuationScenarioReferenceIDRef_, ob);
                    }
                    else if (valuationScenarioReferenceNode.Attributes["href"] != null)
                    {
                        valuationScenarioReferenceIDRef_ = valuationScenarioReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        valuationScenarioReference_ = new ValuationScenarioReference(valuationScenarioReferenceNode);
                    }
                }
                else
                {
                    valuationScenarioReference_ = new ValuationScenarioReference(valuationScenarioReferenceNode);
                }
            }
            
        
            XmlNodeList partialDerivativeNodeList = xmlNode.SelectNodes("partialDerivative");
            
            if (partialDerivativeNodeList != null)
            {
                this.partialDerivative_ = new List<PricingParameterDerivative>();
                foreach (XmlNode item in partialDerivativeNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            partialDerivativeIDRef_ = item.Attributes["id"].Value;
                            partialDerivative_.Add(new PricingParameterDerivative(item));
                            IDManager.SetID(partialDerivativeIDRef_, partialDerivative_[partialDerivative_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            partialDerivativeIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        partialDerivative_.Add(new PricingParameterDerivative(item));
                        }
                    }
                    else
                    {
                        partialDerivative_.Add(new PricingParameterDerivative(item));
                    }
                }
            }
            
        
            XmlNode formulaNode = xmlNode.SelectSingleNode("formula");
            
            if (formulaNode != null)
            {
                if (formulaNode.Attributes["href"] != null || formulaNode.Attributes["id"] != null) 
                {
                    if (formulaNode.Attributes["id"] != null) 
                    {
                        formulaIDRef_ = formulaNode.Attributes["id"].Value;
                        DerivativeFormula ob = new DerivativeFormula(formulaNode);
                        IDManager.SetID(formulaIDRef_, ob);
                    }
                    else if (formulaNode.Attributes["href"] != null)
                    {
                        formulaIDRef_ = formulaNode.Attributes["href"].Value;
                    }
                    else
                    {
                        formula_ = new DerivativeFormula(formulaNode);
                    }
                }
                else
                {
                    formula_ = new DerivativeFormula(formulaNode);
                }
            }
            
        
            XmlNode termNode = xmlNode.SelectSingleNode("term");
            
            if (termNode != null)
            {
                if (termNode.Attributes["href"] != null || termNode.Attributes["id"] != null) 
                {
                    if (termNode.Attributes["id"] != null) 
                    {
                        termIDRef_ = termNode.Attributes["id"].Value;
                        TimeDimension ob = new TimeDimension(termNode);
                        IDManager.SetID(termIDRef_, ob);
                    }
                    else if (termNode.Attributes["href"] != null)
                    {
                        termIDRef_ = termNode.Attributes["href"].Value;
                    }
                    else
                    {
                        term_ = new TimeDimension(termNode);
                    }
                }
                else
                {
                    term_ = new TimeDimension(termNode);
                }
            }
            
        
            XmlNode coordinateNode = xmlNode.SelectSingleNode("coordinate");
            
            if (coordinateNode != null)
            {
                if (coordinateNode.Attributes["href"] != null || coordinateNode.Attributes["id"] != null) 
                {
                    if (coordinateNode.Attributes["id"] != null) 
                    {
                        coordinateIDRef_ = coordinateNode.Attributes["id"].Value;
                        PricingDataPointCoordinate ob = new PricingDataPointCoordinate(coordinateNode);
                        IDManager.SetID(coordinateIDRef_, ob);
                    }
                    else if (coordinateNode.Attributes["href"] != null)
                    {
                        coordinateIDRef_ = coordinateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        coordinate_ = new PricingDataPointCoordinate(coordinateNode);
                    }
                }
                else
                {
                    coordinate_ = new PricingDataPointCoordinate(coordinateNode);
                }
            }
            
        
            XmlNode coordinateReferenceNode = xmlNode.SelectSingleNode("coordinateReference");
            
            if (coordinateReferenceNode != null)
            {
                if (coordinateReferenceNode.Attributes["href"] != null || coordinateReferenceNode.Attributes["id"] != null) 
                {
                    if (coordinateReferenceNode.Attributes["id"] != null) 
                    {
                        coordinateReferenceIDRef_ = coordinateReferenceNode.Attributes["id"].Value;
                        PricingDataPointCoordinateReference ob = new PricingDataPointCoordinateReference(coordinateReferenceNode);
                        IDManager.SetID(coordinateReferenceIDRef_, ob);
                    }
                    else if (coordinateReferenceNode.Attributes["href"] != null)
                    {
                        coordinateReferenceIDRef_ = coordinateReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        coordinateReference_ = new PricingDataPointCoordinateReference(coordinateReferenceNode);
                    }
                }
                else
                {
                    coordinateReference_ = new PricingDataPointCoordinateReference(coordinateReferenceNode);
                }
            }
            
        
        }
        
    
        #region Name_
        private XsdTypeString name_;
        public XsdTypeString Name_
        {
            get
            {
                if (this.name_ != null)
                {
                    return this.name_; 
                }
                else if (this.nameIDRef_ != null)
                {
                    name_ = IDManager.getID(nameIDRef_) as XsdTypeString;
                    return this.name_; 
                }
                else
                {
                      return this.name_; 
                }
            }
            set
            {
                if (this.name_ != value)
                {
                    this.name_ = value;
                }
            }
        }
        #endregion
        
        public string nameIDRef_ { get; set; }
        #region ValuationScenarioReference_
        private ValuationScenarioReference valuationScenarioReference_;
        public ValuationScenarioReference ValuationScenarioReference_
        {
            get
            {
                if (this.valuationScenarioReference_ != null)
                {
                    return this.valuationScenarioReference_; 
                }
                else if (this.valuationScenarioReferenceIDRef_ != null)
                {
                    valuationScenarioReference_ = IDManager.getID(valuationScenarioReferenceIDRef_) as ValuationScenarioReference;
                    return this.valuationScenarioReference_; 
                }
                else
                {
                      return this.valuationScenarioReference_; 
                }
            }
            set
            {
                if (this.valuationScenarioReference_ != value)
                {
                    this.valuationScenarioReference_ = value;
                }
            }
        }
        #endregion
        
        public string valuationScenarioReferenceIDRef_ { get; set; }
        #region PartialDerivative_
        private List<PricingParameterDerivative> partialDerivative_;
        public List<PricingParameterDerivative> PartialDerivative_
        {
            get
            {
                if (this.partialDerivative_ != null)
                {
                    return this.partialDerivative_; 
                }
                else if (this.partialDerivativeIDRef_ != null)
                {
                    return this.partialDerivative_; 
                }
                else
                {
                      return this.partialDerivative_; 
                }
            }
            set
            {
                if (this.partialDerivative_ != value)
                {
                    this.partialDerivative_ = value;
                }
            }
        }
        #endregion
        
        public string partialDerivativeIDRef_ { get; set; }
        #region Formula_
        private DerivativeFormula formula_;
        public DerivativeFormula Formula_
        {
            get
            {
                if (this.formula_ != null)
                {
                    return this.formula_; 
                }
                else if (this.formulaIDRef_ != null)
                {
                    formula_ = IDManager.getID(formulaIDRef_) as DerivativeFormula;
                    return this.formula_; 
                }
                else
                {
                      return this.formula_; 
                }
            }
            set
            {
                if (this.formula_ != value)
                {
                    this.formula_ = value;
                }
            }
        }
        #endregion
        
        public string formulaIDRef_ { get; set; }
        #region Term_
        private TimeDimension term_;
        public TimeDimension Term_
        {
            get
            {
                if (this.term_ != null)
                {
                    return this.term_; 
                }
                else if (this.termIDRef_ != null)
                {
                    term_ = IDManager.getID(termIDRef_) as TimeDimension;
                    return this.term_; 
                }
                else
                {
                      return this.term_; 
                }
            }
            set
            {
                if (this.term_ != value)
                {
                    this.term_ = value;
                }
            }
        }
        #endregion
        
        public string termIDRef_ { get; set; }
        #region Coordinate_
        private PricingDataPointCoordinate coordinate_;
        public PricingDataPointCoordinate Coordinate_
        {
            get
            {
                if (this.coordinate_ != null)
                {
                    return this.coordinate_; 
                }
                else if (this.coordinateIDRef_ != null)
                {
                    coordinate_ = IDManager.getID(coordinateIDRef_) as PricingDataPointCoordinate;
                    return this.coordinate_; 
                }
                else
                {
                      return this.coordinate_; 
                }
            }
            set
            {
                if (this.coordinate_ != value)
                {
                    this.coordinate_ = value;
                }
            }
        }
        #endregion
        
        public string coordinateIDRef_ { get; set; }
        #region CoordinateReference_
        private PricingDataPointCoordinateReference coordinateReference_;
        public PricingDataPointCoordinateReference CoordinateReference_
        {
            get
            {
                if (this.coordinateReference_ != null)
                {
                    return this.coordinateReference_; 
                }
                else if (this.coordinateReferenceIDRef_ != null)
                {
                    coordinateReference_ = IDManager.getID(coordinateReferenceIDRef_) as PricingDataPointCoordinateReference;
                    return this.coordinateReference_; 
                }
                else
                {
                      return this.coordinateReference_; 
                }
            }
            set
            {
                if (this.coordinateReference_ != value)
                {
                    this.coordinateReference_ = value;
                }
            }
        }
        #endregion
        
        public string coordinateReferenceIDRef_ { get; set; }
        public string choiceStr_0;
        public string choiceStr_1;
        public string choiceStr_2;
        
    
        
    
    }
    
}

